src/main/java/com/mongodb/jdbc/MongoFunctions.java

```diff
@@ -0,0 +1,110 @@
+package com.mongodb.jdbc;
+
+public abstract class MongoFunctions {
+    public enum FunctionCategory {
+        STRING_FUNC,
+        NUM_FUNC,
+        TIME_DATE_FUNC,
+        SYSTEM_FUNC,
+        CONV_FUNC,
+        UNCATEGORIZED_FUNC
+    }
+
+    public static class MongoFunction {
+        public String name;
+        public String returnType;
+        public String comment;
+        public String[] argTypes;
+        public FunctionCategory functionCategory;
+
+        protected MongoFunction(
+                String name,
+                String returnType,
+                String comment,
+                String[] argTypes,
+                FunctionCategory category) {
+            this.name = name;
+            this.returnType = returnType;
+            this.comment = comment;
+            this.argTypes = argTypes;
+            this.functionCategory = category;
+        }
+
+        protected MongoFunction(String name, String returnType, String comment, String[] argTypes) {
+            this.name = name;
+            this.returnType = returnType;
+            this.comment = comment;
+            this.argTypes = argTypes;
+            this.functionCategory = FunctionCategory.UNCATEGORIZED_FUNC;
+        }
+    }
+
+    public MongoFunction[] functions;
+    public String numericFunctionsString;
+    public String stringFunctionsString;
+    public String dateFunctionsString;
+    public String systemFunctionsString;
+
+    protected MongoFunctions(MongoFunction[] functions) {
+        this.functions = functions;
+        initCategorizedFunctionsList();
+    }
+
+    // Build the list of numeric, string, dateTime and system functions.
+    private void initCategorizedFunctionsList() {
+        StringBuilder numericFunctionBuilder = new StringBuilder();
```

Comment:
nit: naming goes from numericFunction`s`String to numericFunctionBuilder without the 's'.  Suggest adding the 's' for consistency.